src/Xamarin.Android.Build.Tasks/Utilities/XAJavaTypeScanner.cs

```diff
@@ -1,103 +1,91 @@
 using System;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
+using System.IO;
 
 using Java.Interop.Tools.Cecil;
+using Microsoft.Android.Build.Tasks;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
 using Mono.Cecil;
 using Xamarin.Android.Tools;
 
 namespace Xamarin.Android.Tasks;
 
-class JavaType
-{
-	public readonly TypeDefinition Type;
-	public readonly IDictionary<AndroidTargetArch, TypeDefinition>? PerAbiTypes;
-	public bool IsABiSpecific { get; }
-
-	public JavaType (TypeDefinition type, IDictionary<AndroidTargetArch, TypeDefinition>? perAbiTypes)
-	{
-		Type = type;
-		if (perAbiTypes != null) {
-			PerAbiTypes = new ReadOnlyDictionary<AndroidTargetArch, TypeDefinition> (perAbiTypes);
-			IsABiSpecific = perAbiTypes.Count > 1 || (perAbiTypes.Count == 1 && !perAbiTypes.ContainsKey (AndroidTargetArch.None));
-		}
-	}
-}
-
 class XAJavaTypeScanner
 {
-	sealed class TypeData
-	{
-		public readonly TypeDefinition FirstType;
-		public readonly Dictionary<AndroidTargetArch, TypeDefinition> PerAbi;
-
-		public bool IsAbiSpecific => !PerAbi.ContainsKey (AndroidTargetArch.None);
-
-		public TypeData (TypeDefinition firstType)
-		{
-			FirstType = firstType;
-			PerAbi = new Dictionary<AndroidTargetArch, TypeDefinition> ();
-		}
-	}
+	// Names of assemblies which don't have Mono.Android.dll references, or are framework assemblies, but which must
+	// be scanned for Java types.
+	static readonly HashSet<string> SpecialAssemblies = new HashSet<string> (StringComparer.OrdinalIgnoreCase) {
```

Comment:
Why do these assemblies need to be special-cased?  They weren't before…

Reply:
The change was made to scan less assemblies, ignore the ones that are framework assemblies or don't have `Mono.Android` references.

This speeds up scanning in larger projects, but it ignored `Mono.Android*.dll` and `Java.Interop.dll`. I don't remember why the latter
had to be included, it was probably because that it fixed some failing tests. I'll remove it and push the change, we'll see if anything
fails because of that.